library/src/interactor_impl.cxx

```diff
@@ -408,6 +400,34 @@ class interactor_impl::internals
     this->VTKInteractor->ExitCallback();
   }
 
+  void AnimateCameraTransition(const std::function<void(camera&, double)>& update_camera)
```

Comment:
I'm not completely convinced by the API, could you explain the rationale ?

Reply:
The idea is to provide a generic way to transition the camera from one position to the next without having to know too much about the animation itself as that would be handled by the app. 

So here the calling code provides a function that modifies a given camera according to an interpolant ratio in `[0,1]` where `0` is the original position and `1` is the final position.
Then the app will call that function at the appropriate time(s) with the correct ratio value. Right now we call it as fast as we can render but we could include rate limiting, callbacks on each frames, change the duration, easing, ... later without the calling code needing to know.

If the issue is the code itself we could go with `void AnimateCameraTransition(const std::function<camera_state_t(double interpolant)>& interpolate_state)` for a more Functional Programming (pretty! no side effects!) style at the cost of having to build the state object every frame instead of mutating the camera directly.

Not sure that answers your concern, let me know

Reply:
> Do you think it will happen often that one will want to redefined how the interpolation should be done ?

I can see 3 cases that would realistically come up short term:
- the (only) case we have currently is moving the focal point by picking, that's all xyz linear interpolation.
- there's been talk about snapping to nearest axis (or three-quarter view) and that could be done with spherical coords (you'd lerp the radius and angles before going back to xyz)
- if you wanted to animate the current "reset default view" from an arbitrary position you'd probably want quaternions to do it right

Reply:
thats fair